src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -605,6 +607,23 @@ void Graph::AllocateWithReuse() {
                 erase = true;
                 break;
             }
+
+            // Special allocation for string tensors
+            if (edge->getDesc().getPrecision() == element::string && edge->getStatus() == Edge::Status::NeedAllocation &&
+                    !edge->getParent()->isConstant()) {
+                auto memory = std::make_shared<StringMemory>(getEngine(), edge->getDesc());
+                edge->reuse(memory);
+                for (auto& edge_c : cluster) {
+                    if (edge_c->getDesc().getPrecision() == element::string && edge_c->getStatus() == Edge::Status::NeedAllocation &&
+                            !edge_c->getParent()->isConstant()) {
+                        auto memory = std::make_shared<StringMemory>(getEngine(), edge_c->getDesc());
+                        edge_c->reuse(memory);
+                    }
```

Comment:
We need add an assertion that all the edges in cluster have `string` precision. Just as a sanity check.

Reply:
Done